### ISSUE TITLE
Using UFF made the result worse?

### ISSUE THREAD

**Issue by magicrane** [nomewang/M3DM#36]
Whats excellent codes and thanks for author's work! But when i tried to repitition its looks like i'm in trouble:
![image](https://github.com/user-attachments/assets/b60955d8-2722-411e-8d52-92facce71c02)
I followed the markdown step by step and in the last phrase, an error occured in results that i found using the trained UFF made the result worse. Did anyone meet this problem?

**Comment by hzzzzzhappy** [comment]
i meet a similar situation. But i think my training is not enough, when i use the paper result I can get the same result

**Comment by magicrane** [comment]
> i meet a similar situation. But i think my training is not enough, when i use the paper result I can get the same result
Thanks for your reply!!! Do you mean that  I have to train the UFF for 750 eppchs? I'll try it later. In fact training without UFF gave me a better result than that mentions in paper with UFF. I tried for several times but the results were same.

**Comment by hzzzzzhappy** [comment]
> > 我遇到了类似的情况。但是我觉得我的训练还不够，当我使用论文结果时，我可以得到相同的结果感谢您的回复!!你的意思是我必须训练 750 个 eppch 的 UFF 吗？我稍后再试。事实上，没有 UFF 的训练比论文中提到的有 UFF 的训练给我的效果要好。我尝试了几次，但结果都是一样的。

thanks for u reply.  I think training a full set of epochs is necessary because contrastive learning is not stable, which may lead to some unexpected errors. We're actually working on improving that, and if you have more questions, we can talk more.

**Comment by magicrane** [comment]
> thanks for u reply. I think training a full set of epochs is necessary because contrastive learning is not stable, which may lead to some unexpected errors. We're actually working on improving that, and if you have more questions, we can talk more.
Hi @hzzzzzhappy ! Sorry to bother you again with the same problem..
when running this command:

```
python3 main.py \
--method_name DINO+Point_MAE \
--memory_bank multiple \
--rgb_backbone_name vit_base_patch8_224_dino \
--xyz_backbone_name Point_MAE \
--save_feature
```

I got the result showing as follow
![image](https://github.com/user-attachments/assets/a1156e8e-6a3e-4421-9cd6-3385de461348)

However, when I running this

```
python3 main.py \
--method_name DINO+Point_MAE+Fusion \
--use_uff \
--memory_bank multiple \
--rgb_backbone_name vit_base_patch8_224_dino \
--xyz_backbone_name Point_MAE \
--fusion_module_path checkpoints/checkpoint-749.pth 

```

I got this worse result .
![image](https://github.com/user-attachments/assets/e94e4d29-6fde-4c25-8b35-b6c19f6ed93c)
I am vexed by the inability to obtain the results of the article.  

training the UFF, I only modified the parameter of epochs:

`UFFOMP_NUM_THREADS=1 python3 -m torch.distributed.run --nproc_per_node=1 fusion_pretrain.py    --accum_iter 16 --lr 0.003 --batch_size 16 --data_path datasets/patch_lib --output_dir checkpoints `

What am I supposed to do then? Looking forward to your reply.

**Comment by hzzzzzhappy** [comment]
Hi!  Maybe you have some difficult results, I apologize for the late reply, maybe you can directly email to me: 2023362051@email.szu.edu.cn.

We haven't had this problem, but as a rule of thumb, there are a few issues:

1) Have you done the preprocessing correctly? The preprocessing of Mvtec3D-AD is crucial to the results.

2) Do you change the weighting coefficients? 3D, RGB and fusion coefficients are very important to the result. If you accidentally change this, the result will be bad.

3) Are there any issues with your version? But as a rule of thumb, version doesn't matter that much.


Since I am not the author of this paper, my advice is based on my own experience and I hope it will be helpful to you. But I'm more of a data preprocessing problem.

**Comment by magicrane** [comment]
> Hi! Maybe you have some difficult results, I apologize for the late reply, maybe you can directly email to me: [2023362051@email.szu.edu.cn](mailto:2023362051@email.szu.edu.cn).
> 
> We haven't had this problem, but as a rule of thumb, there are a few issues:
> 
> 1. Have you done the preprocessing correctly? The preprocessing of Mvtec3D-AD is crucial to the results.
> 2. Do you change the weighting coefficients? 3D, RGB and fusion coefficients are very important to the result. If you accidentally change this, the result will be bad.
> 3. Are there any issues with your version? But as a rule of thumb, version doesn't matter that much.
> 
> Since I am not the author of this paper, my advice is based on my own experience and I hope it will be helpful to you. But I'm more of a data preprocessing problem.

Thank you for your explanation. It is quite possible that the issue lies in my settings. I will thoroughly review them to see if there is anything I have overlooked. Once again, I sincerely appreciate your patient response!

**Comment by huanxi-yashi** [comment]
@magicrane 您好 请问您的问题解决了吗 请问是如何解决的呢？我遇到了和您一样的问题

**Comment by hzzzzzhappy** [comment]
> [@magicrane](https://github.com/magicrane) 您好 请问您的问题解决了吗 请问是如何解决的呢？我遇到了和您一样的问题

Hi, maybe you met the problem. I remember I modified part of the code earlier to make a selection by storing the top 30 weights with the least amount of loss. 

I reproduced slightly worse results, but the error is acceptable.

I hope you are able to reproduce this work and also observe some of the latest work if you can.

**Comment by magicrane** [comment]
> [@magicrane](https://github.com/magicrane) 您好 请问您的问题解决了吗 请问是如何解决的呢？我遇到了和您一样的问题

Well, sorry to tell you that I failed to reproduce the result by using uff training on my device, But I got the right result using the pretrain weighs provided by author. Possible differences in results may be caused by different devices.
